modules/gapi/src/backends/fluid/gfluidbackend.cpp

```diff
@@ -658,7 +658,22 @@ void cv::gimpl::GFluidExecutable::initBufferRois(std::vector<int>& readStarts,
                     }
 
                     int readStart = resized.y;
-                    cv::gapi::own::Rect roi = adjFilterRoi(resized, fd.border_size, in_meta.size.height);
+                    auto roi = adjFilterRoi(resized, fd.border_size, in_meta.size.height);
+
+                    // Extend required input roi to be even if it's produced by NV12toRGB
+                    if (!in_node->inNodes().empty()) {
+                        auto in_data_producer = in_node->inNodes().front();
+                        if (fg.metadata(in_data_producer).get<FluidUnit>().k.m_kind == GFluidKernel::Kind::NV12toRGB) {
+                            if (roi.y % 2 != 0) {
+                                roi.y--;
+                                roi.height++;
+                            }
+
+                            if (roi.height % 2 != 0) {
+                                roi.height++;
```

Comment:
What is about a check with fitting into image?
Why is not to shrink `roi`?

Reply:
This is actually done later via adjFilterRoi call, I was thinking to clarify this step, will do